### PR TITLE
Use built-in AMI bootstrapping scripts for unmanaged nodes

### PR DESCRIPTION
**Pull request by Callisto13**
### Description

Closes https://github.com/weaveworks/eksctl/issues/3464.

This PR removes all custom nodebootstrapping for Ubuntu and AL2 unmanaged nodes. We will now use the built-in `/etc/eks/bootstrap.sh` for all our bootstrapping needs. The new implementation _should_ have complete parity with the old one; users should not notice that anything has changed.

Also: 
- I have tidied-up the Windows and Bottlerocket bootstrap code and tests so that they fit the new interface. Windows was already using the built-in bootstrapper. Bottlerocket is doing something else which I don't want to get into.
- Speaking of interfaces, the NodeGroupResourceSet now contains a "bootstrapper". This decouples the cfn builder from nodebootstrapper, and means cfn no longer unit-tests userdata generation. (That test file is still a disaster, it is on my list)
- I have not touched Managed nodegroup bootstrapping as that is using the built-in bootstrapper, and this diff was already large enough.
- Deleted so much stuff!

See https://github.com/weaveworks/eksctl/issues/3464#issuecomment-810135117 for more info on decisions.

### Checklist
- [x] Added tests that cover your change (if possible)
- [x] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [x] Manually tested
- [x] Made sure the title of the PR is a good description that can go into the release notes
- [x] (Core team) Added labels for change area (e.g. `area/nodegroup`) and kind (e.g. `kind/improvement`)

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [x] Backfilled missing tests for code in same general area :tada:
- [x] Refactored something and made the world a better place :star2:

### CODE REVIEW

**Comment by aclevername** [comment]
have you run the integration tests on this branch? did it pass?

**Comment by Callisto13** [comment]
integration tests all good 👍

File: pkg/nodebootstrap/al2_test.go [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+
+var _ = Describe("AmazonLinux2 User Data", func() {
+	var (
+		clusterName  string
+		ng           *api.NodeGroup
+		bootstrapper *nodebootstrap.AmazonLinux2
+	)
+
+	BeforeEach(func() {
+		clusterName = "something-awesome"
```
Comment (aclevername) [inline pkg/nodebootstrap/al2_test.go:19]: AWESOMEEE

File: pkg/nodebootstrap/assets/bootstrap.al2.sh [hunk 2]
```diff
@@ -4,42 +4,26 @@ set -o errexit
...
+  --dns-cluster-ip "${CLUSTER_DNS}" \
+  --kubelet-extra-args "--register-with-taints=${NODE_TAINTS} --node-labels=${NODE_LABELS}"
 
-NODE_IP="$(get_metadata local-ipv4)"
-INSTANCE_ID="$(get_metadata instance-id)"
-INSTANCE_TYPE="$(get_metadata instance-type)"
-AWS_SERVICES_DOMAIN="$(get_metadata services/domain)"
+echo "eksctl: merging user options into kubelet-config.json"
+trap 'rm -f ${TMP_KUBE_CONF}' EXIT
+jq -s '.[0] * .[1]' "${KUBELET_CONFIG}" "${KUBELET_EXTRA_ARGS}" > "${TMP_KUBE_CONF}"
```
Comment (aclevername) [inline pkg/nodebootstrap/assets/bootstrap.al2.sh:16]: that was surprisingly simple

File: pkg/nodebootstrap/windows.go [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+func NewWindowsBootstrapper(clusterName string, ng *api.NodeGroup) *Windows {
+	return &Windows{
+		clusterName: clusterName,
+		ng:          ng,
+	}
+}
+
+func (b *Windows) UserData() (string, error) {
+	bootstrapScript := `<powershell>
+[string]$EKSBootstrapScriptFile = "$env:ProgramFiles\Amazon\EKS\Start-EKSBootstrap.ps1"
```
Comment (aclevername) [inline pkg/nodebootstrap/windows.go:28]: :vomiting_face:

**Review by aclevername — APPROVED** [review]
LGTM, big diff so I won't claim to understand all the changes, but with a lot of new test coverage added and the integration tests passing I'm happy :+1: :tada:

File: pkg/nodebootstrap/nodebootstrap_suite_test.go [hunk 4]
```diff
@@ -1,11 +1,21 @@
-package nodebootstrap
+package nodebootstrap_test
```
[the remark this answers is not in the document]
Comment (Callisto13) [reply pkg/nodebootstrap/nodebootstrap_suite_test.go:1]: ikr

**Comment by Callisto13** [comment]
This has been reverted temporarily as we safeguard around a breaking change.

**Pull request merged**